beets/dbcore/query.py

```diff
@@ -163,14 +169,18 @@ class StringFieldQuery(FieldQuery):
     """
 
     @classmethod
-    def value_match(cls, pattern, value):
+    def value_match(cls: Type['StringFieldQuery'], pattern: str, value: Any):
```

Comment:
Let's just drop the type for `cls` here?  I suppose type checkers don't need it, and if we really need to type `cls`, I think we should wait until we drop support for older Python which doesn't have the `Self` type yet. (Same for `cls` annotations below.)